app/integration_action.go

```diff
@@ -140,6 +127,50 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 		upstreamURL = action.Integration.URL
 	}
 
+	// get channel, user and team details
+	chanChan := make(chan store.StoreResult, 1)
+	go func() {
+		channel, err := a.Srv.Store.Channel().Get(upstreamRequest.ChannelId, true)
+		chanChan <- store.StoreResult{Data: channel, Err: err}
+		close(chanChan)
+	}()
+
+	userChan := make(chan store.StoreResult, 1)
```

Comment:
Can you please move this code to the other request above? This fetches the data earlier to save time.

Reply:
It's not possible, this request makes use of `upstreamRequest.ChannelId`, initialized differently on line 85 (from the cookie) and line 102 (from the post).
As such, it depends on the result of pchan and cannot run asynchronously to it.

Reply:
You are right. :+1: